pennylane/ops/op_math/prod.py

```diff
@@ -276,32 +286,56 @@ def diagonalizing_gates(self):
         Returns:
             list[.Operator] or None: a list of operators
         """
-
-        eigen_vectors = self.eigendecomposition["eigvec"]
-        return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        if self.has_overlapping_wires:
+            eigen_vectors = self.eigendecomposition["eigvec"]
+            return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+        diag_gates = []
+        for factor in self.factors:
+            diag_gates.extend(factor.diagonalizing_gates())
+        return diag_gates
 
     def eigvals(self):
-        r"""Return the eigenvalues of the specified operator.
+        """Return the eigenvalues of the specified operator.
 
         This method uses pre-stored eigenvalues for standard observables where
         possible and stores the corresponding eigenvectors from the eigendecomposition.
 
         Returns:
             array: array containing the eigenvalues of the operator
         """
-        return self.eigendecomposition["eigval"]
+        if self.has_overlapping_wires:
+            return self.eigendecomposition["eigval"]
+        eigvals = [
+            qml.utils.expand_vector(factor.eigvals(), list(factor.wires), list(self.wires))
+            for factor in self.factors
+        ]
+
+        return qml.math.prod(eigvals, axis=0)
 
     def matrix(self, wire_order=None):
         """Representation of the operator as a matrix in the computational basis."""
+
+        sorted_factors = self.factors
```

Comment:
change name since these are no longer being sorted

Reply:
Removed this line. Now I use `self.factors` directly.